File: CS/WebSite/WebUserControl.ascx.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using DevExpress.Web.ASPxNavBar;

public partial class WebUserControl : System.Web.UI.UserControl, ITemplate
{

    protected void Page_Load(object sender, EventArgs e){
        NavBarItemTemplateContainer container = this.NamingContainer as NavBarItemTemplateContainer;
        string groupIndex = container.Item.Group.Index.ToString();
        ASPxButton1.ClientSideEvents.Click = "function(s, e){OnButtonClick(s, e, true, " + groupIndex + ")}";
        ASPxButton2.ClientSideEvents.Click = "function(s, e){OnButtonClick(s, e, false, " + groupIndex + ")}";
    }

    void ITemplate.InstantiateIn(Control container) {
        container.Controls.Add(this);
    }
}

File: VB/WebSite/WebUserControl.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Collections
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports DevExpress.Web.ASPxNavBar

Partial Public Class WebUserControl
	Inherits System.Web.UI.UserControl
	Implements ITemplate

	Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
		Dim container As NavBarItemTemplateContainer = TryCast(Me.NamingContainer, NavBarItemTemplateContainer)
		Dim groupIndex As String = container.Item.Group.Index.ToString()
		ASPxButton1.ClientSideEvents.Click = "function(s, e){OnButtonClick(s, e, true, " & groupIndex & ")}"
		ASPxButton2.ClientSideEvents.Click = "function(s, e){OnButtonClick(s, e, false, " & groupIndex & ")}"
	End Sub

	Private Sub InstantiateIn(ByVal container As Control) Implements ITemplate.InstantiateIn
		container.Controls.Add(Me)
	End Sub
End Class
